### PR TITLE
1.0/0.0 == NULL, 1 div 0 == NULL

### PR DESCRIPTION
**Pull request by andrew-wm-arthur**
Signed-off-by: Andy Arthur <andy@liquidata.co>

### CODE REVIEW

File: sql/expression/arithmetic_test.go [hunk 1]
```diff
@@ -146,13 +146,14 @@ func TestDiv(t *testing.T) {
...
-		{"-1 / 1", -1, 1, -1},
-		{"0 / 1234567890", 0, 12345677890, 0},
-		{"3.14159 / 3.0", 3.14159, 3.0, float64(3.14159) / float64(3.0)},
-		{"1/0", 1, 0, math.Inf(1)},
-		{"-1/0", -1, 0, math.Inf(-1)},
+		{"1 / 1", 1, 1, 1, false},
+		{"-1 / 1", -1, 1, -1, false},
+		{"0 / 1234567890", 0, 12345677890, 0, false},
+		{"3.14159 / 3.0", 3.14159, 3.0, float64(3.14159) / float64(3.0), false},
+		{"1/0", 1, 0, math.Inf(1), true},
```
Comment (zachmu) [inline sql/expression/arithmetic_test.go:155]: These are wrong

File: sql/expression/arithmetic_test.go [hunk 2]
```diff
@@ -352,11 +357,13 @@ func TestIntDiv(t *testing.T) {
...
 	}{
-		{"1 div 1", 1, 1, 1},
-		{"8 div 3", 8, 3, 2},
-		{"1 div 3", 1, 3, 0},
-		{"0 div -1024", 0, -1024, 0},
+		{"1 div 1", 1, 1, 1, false},
+		{"8 div 3", 8, 3, 2, false},
+		{"1 div 3", 1, 3, 0, false},
+		{"0 div -1024", 0, -1024, 0, false},
+		{"1 div 0", 1, 0, 0, true},
```
Comment (zachmu) [inline sql/expression/arithmetic_test.go:366]: 0 div 0 would be a good test too (NULL(

File: sql/expression/arithmetic_test.go [hunk 3]
```diff
@@ -146,13 +146,14 @@ func TestDiv(t *testing.T) {
...
-		{"0 / 1234567890", 0, 12345677890, 0},
-		{"3.14159 / 3.0", 3.14159, 3.0, float64(3.14159) / float64(3.0)},
-		{"1/0", 1, 0, math.Inf(1)},
-		{"-1/0", -1, 0, math.Inf(-1)},
+		{"1 / 1", 1, 1, 1, false},
+		{"-1 / 1", -1, 1, -1, false},
+		{"0 / 1234567890", 0, 12345677890, 0, false},
+		{"3.14159 / 3.0", 3.14159, 3.0, float64(3.14159) / float64(3.0), false},
+		{"1/0", 1, 0, math.Inf(1), true},
+		{"-1/0", -1, 0, math.Inf(-1), true},
```
Comment (zachmu) [inline sql/expression/arithmetic_test.go:156]: 0.0 / 0.0 is another good test case

File: sql/expression/arithmetic_test.go (hunk 1 above)
Comment (andrew-wm-arthur) [reply sql/expression/arithmetic_test.go]: I changed the expected val, but it's ignored if null is true

**Pull request merged**